tests/integration/s2n_dynamic_record_size_test.py

```diff
@@ -200,54 +199,40 @@ def test(host, port, test_ciphers, threshold):
 
     return failed
 
-def analyze_latency_dump(array):
-    failed = 0
+
+def analyze_tcp_dump(array, threshold):
+    failed = 1
+    packets_transferred = 0
+    array_len = len(array)
+    # get the mss from first message
     mss = get_local_mtu() - 40
     first_line = array[0]
-    if ("mss" in first_line):
+    if "mss" in first_line:
         mss_pos = first_line.find("mss")
         mss_str = first_line[mss_pos : mss_pos + 10]
         mss = int(mss_str[4 : mss_str.find(',')])
     else:
-        print ("use default mss")
-    # print("mss={}".format(mss))
-    
-    for i in range(0, 18):
-        output = array[i]
-        # print(output)
-        pos = output.find("length")
-        if pos < 0:
-            continue
-        length = output[pos + 6 : len(output)]
-        # Tcp package size should always <= mss
-        if int(length) > mss:
-            failed = 1
-            break
-
-    return failed
-
-def analyze_throughput_dump(array):
-    failed = 1
-    array_len = len(array)
+        print ("using default mss")
 
-    for i in range(18, 36):
-        if i >= array_len:
-            print("Array len is ", array_len, ", expecting >= ", i)
-            print(array)
-            return failed
-        output = array[i]
-        # print(output)
-        pos = output.find("length")
+    for i in range(0, array_len):
+        # record the length of each packet in TCP dump
+        pos = array[i].find("length")
         if pos < 0:
             continue
-        length = output[pos + 6 : len(output)]
-        # Tcp package size can exceed MTU, which results in segementation
-        if int(length) > 1500:
-            failed = 0
+        length = array[i][pos + 6 : len(array[i])]
+        packets_transferred += int(length)
+        # print ("Packet:",i, "packet size:", length, "packets transferred:",packets_transferred, "threshold met:", packets_transferred > threshold)
+        # optimized for latency - before the threshold has been met, the TCP packet size should always <= mss
+        if packets_transferred < threshold and int(length) > mss:
+            break
```

Comment:
This tripped me up on the first reading, can you expand/add a comment that if you get here not enough low latency data was sent and the test failed.

Reply:
Addressed in [#1e2620f](https://github.com/awslabs/s2n/pull/2292/commits/1e2620fc4a7e9085f5b52b9796bc2799ce3edc93)